src/main/java/org/opensrp/web/rest/ClientResource.java

```diff
@@ -150,6 +150,12 @@ public Client update(Client entity) {//TODO check if send property and id matche
 		searchBean.setAttributeType(StringUtils.isBlank(attributes) ? null : attributes.split(":", -1)[0]);
 		searchBean.setAttributeValue(StringUtils.isBlank(attributes) ? null : attributes.split(":", -1)[1]);
 
+		String locationIds = getStringFilter("locationIds", request);
+		List<String> locations = StringUtils.isBlank(locationIds) ? new ArrayList<>() :
+				Arrays.asList(locationIds.split(","));
+		if (locations.size() != 0) {
+			searchBean.setLocations(locations);
+		}

```

Comment:
This seems redundant, having initialize an empty array and the checking  if array is empty.

```suggestion
		String locationIds = getStringFilter("locationIds", request);
		if( StringUtils.isNotBlank(locationIds))){
		       	searchBean.setLocations(Arrays.asList(locationIds.split(",")));
		}

```